Gems/Atom/RPI/Code/Source/RPI.Public/RPIUtils.cpp

```diff
@@ -222,5 +419,119 @@ namespace AZ
         {
             return GetComputeShaderNumThreads(shaderAsset, &dispatchDirect.m_threadsPerGroupX, &dispatchDirect.m_threadsPerGroupY, &dispatchDirect.m_threadsPerGroupZ);
         }
+
+        template<>
+        float GetSubImagePixelValue<float>(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, uint32_t x, uint32_t y, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+        {
+            return Internal::GetSubImagePixelValueInternal<float>(imageAsset, x, y, componentIndex, mip, slice);
+        }
+
+        template<>
+        AZ::u32 GetSubImagePixelValue<AZ::u32>(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, uint32_t x, uint32_t y, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+        {
+            return Internal::GetSubImagePixelValueInternal<AZ::u32>(imageAsset, x, y, componentIndex, mip, slice);
+        }
+
+        template<>
+        AZ::s32 GetSubImagePixelValue<AZ::s32>(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, uint32_t x, uint32_t y, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+        {
+            return Internal::GetSubImagePixelValueInternal<AZ::s32>(imageAsset, x, y, componentIndex, mip, slice);
+        }
+
+        void GetSubImagePixelValues(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::span<float> outValues, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+        {
+            // TODO: Use the component index
+            (void)componentIndex;
+
+            if (!imageAsset.IsReady())
+            {
+                return;
+            }
+
+            auto imageData = imageAsset->GetSubImageData(mip, slice);
+
+            if (!imageData.empty())
+            {
+                const AZ::RHI::ImageDescriptor imageDescriptor = imageAsset->GetImageDescriptor();
+                auto width = imageDescriptor.m_size.m_width;
+                const uint32_t pixelSize = AZ::RHI::GetFormatSize(imageDescriptor.m_format);
+
+                size_t outValuesIndex = 0;
+                for (uint32_t y = topLeft.second; y < bottomRight.second; ++y)
+                {
+                    for (uint32_t x = topLeft.first; x < bottomRight.first; ++x)
+                    {
+                        size_t imageDataIndex = (y * width + x) * pixelSize;
+
+                        auto& outValue = outValues[outValuesIndex++];
+                        outValue = Internal::RetrieveFloatValue(imageData.data(), imageDataIndex, imageDescriptor.m_format);
+                    }
+                }
+            }
+        }
+
+        void GetSubImagePixelValues(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::span<AZ::u32> outValues, uint32_t componentIndex, uint32_t mip, uint32_t slice)
```

Comment:
Maybe we can return boolean for these functions in case the the get functions didn't get the desired data.

Reply:
Will do!